### PR TITLE
add Azure Pipelines Scaler doc

### PR DESCRIPTION
**Pull request by troydn**
This PR adds documentation for Azure Pipelines Scaler.

Relates to https://github.com/kedacore/keda/issues/1705

### CODE REVIEW

File: content/docs/2.3/scalers/azure-pipelines.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
++++
+title = "Azure Pipelines"
+layout = "scaler"
+availability = "v2.3+"
+maintainer = "Community"
+description = "Scale applications based on Azure Pipelines Queues."
```
Comment (tomkerkhove) [inline content/docs/2.3/scalers/azure-pipelines.md:6]:
```suggestion
description = "Scale applications based on agent pool queues for Azure Pipelines."
```

File: content/docs/2.3/scalers/azure-pipelines.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+go_file = "azure_pipelines_scaler"
++++
+
+### Trigger Specification
+
+This specification describes the `azure-pipelines` trigger for Azure Pipelines Queue. It scales based on the queue length of Azure DevOps jobs in a given Agent Pool.
```
Comment (tomkerkhove) [inline content/docs/2.3/scalers/azure-pipelines.md:12]:
```suggestion
This specification describes the `azure-pipelines` trigger for Azure Pipelines. It scales based on the amount of pipeline runs pending in a given agent pool.
```

File: content/docs/2.3/scalers/azure-pipelines.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      # Optional: Target queue length
+      targetPipelinesQueueLength: "1" # Default 1
+    authenticationRef:
+     name: pipeline-trigger-auth
+```
+
+**Parameter list:**
+
+- `poolID` - Id of the queue.
+- `organizationURLFromEnv` - Name of the environment variable your deployment uses to get the organizationURL string.
```
Comment (tomkerkhove) [inline content/docs/2.3/scalers/azure-pipelines.md:33]:
```suggestion
- `organizationURLFromEnv` - Name of the environment variable your deployment uses to get the URL for your Azure DevOps organization.
```

File: content/docs/2.3/scalers/azure-pipelines.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+- `personalAccessTokenFromEnv` - Name of the environment variable your deployment uses to get the personalAccessToken string.
```
Comment (tomkerkhove) [inline content/docs/2.3/scalers/azure-pipelines.md:34]: Do you know what permissions are required? We should include this here as well.
```suggestion
- `personalAccessTokenFromEnv` - Name of the environment variable that provides the personal access token (PAT) for Azure DevOps. Learn more how to create on [in the official docs](https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops&tabs=preview-page).
```

File: content/docs/2.3/scalers/azure-pipelines.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+- `targetPipelinesQueueLength` - Target value for queue length passed to the scaler. Example: if one pod can handle 10 jobs, set the queue length target to 10. If the actual number of jobs in the queue is 30, the scaler scales to 3 pods. (default: 1)
```
Comment (tomkerkhove) [inline content/docs/2.3/scalers/azure-pipelines.md:35]:
```suggestion
- `targetPipelinesQueueLength` - Target value for the amount of pending jobs in the queue to scale on. (default: 1)
  - Example - If one pod can handle 10 jobs, set the queue length target to 10. If the actual number of jobs in the queue is 30, the scaler scales to 3 pods.
```

File: content/docs/2.3/scalers/azure-pipelines.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+### Authentication Parameters
+
+As an alternative to using environment variables, you can authenticate with Azure Devops using a Personal Access Token via `TriggerAuthentication` configuration.
+
+**Personal Access Token Authentication:**
+
+- `organizationURL` - The Azure DevOps organization
```
Comment (tomkerkhove) [inline content/docs/2.3/scalers/azure-pipelines.md:43]:
```suggestion
- `organizationURL` - The URL of the Azure DevOps organization
```

File: content/docs/2.3/scalers/azure-pipelines.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+- `personalAccessToken` - The Azure DevOps Personal Access Token
```
Comment (tomkerkhove) [inline content/docs/2.3/scalers/azure-pipelines.md:44]:
```suggestion
- `personalAccessToken` - The Personal Access Token (PAT) for Azure DevOps
```

**Review by tomkerkhove — COMMENTED** [review]
LGTM, added a few editorial suggestions though.

**Review by tomkerkhove — APPROVED** [review]
LGTM, just waiting for the feature PR to be merged!

**Review by zroubalik — APPROVED** [review]

**Comment by tomkerkhove** [comment]
@troydn I think this is a solid case for a blog post talking about it end-to-end. Are you willing to write one that we publish on keda.sh where a self-hosted agent is deployed and we add autoscaling to it?

**Comment by troydn** [comment]
> @troydn I think this is a solid case for a blog post talking about it end-to-end. Are you willing to write one that we publish on keda.sh where a self-hosted agent is deployed and we add autoscaling to it?

@tomkerkhove Yes, I could write one. Should I create a seperate PR for this?

**Comment by tomkerkhove** [comment]
A seperate PR would be great, thanks!

What I was thinking about was:
- Deploy self-hosted agent
- Schedule some pipeline runs
- Apply autoscaling
- Show outcome
Would that make sense to you?

**Pull request merged**